src/client/debugger/jupyter/debuggingManager.ts

```diff
@@ -92,102 +93,119 @@ export class DebuggingManager implements IExtensionSingleActivationService, IDeb
                 createDebugAdapterDescriptor: async (session) => this.createDebugAdapterDescriptor(session)
             }),
 
-            this.commandManager.registerCommand(DSCommands.DebugNotebook, async () => {
-                const editor = this.vscNotebook.activeNotebookEditor;
-                if (editor) {
-                    if (await this.checkForIpykernel6(editor.document)) {
-                        this.updateToolbar(true);
-                        void this.startDebugging(editor.document);
+            this.commandManager.registerCommand(
+                DSCommands.DebugNotebook,
+                throttle(async () => {
+                    const editor = this.vscNotebook.activeNotebookEditor;
```

Comment:
I think the solution is, if we're already debugging or have started then return from here.
I'd if in progress then get out, mashed three throttle unnecessary.